landing-zone/f5_config/variables.tf

```diff
@@ -0,0 +1,203 @@
+
+variable "region" {
+  description = "The VPC region to instantiate the F5 BIG-IP instance"
+  type        = string
+}
+
+##################################################################################
+
+
+##################################################################################
+# VPC Variables
+##################################################################################
+
+variable "vpc_id" {
+  description = "The VPC ID where the instance will be provisioned"
+  type        = string
+}
+
+variable "zone" {
+  description = "Zone where the instance will be provisioned"
+  type        = string
+}
+
+variable "secondary_subnets" {
+  description = "List of secondary network interfaces to add to vsi secondary subnets must be in the same zone as VSI. This is only recommended for use with a deployment of 1 VSI."
+  type = list(
+    object({
+      name = string
+      id   = string
+      zone = string
+      cidr = string
+    })
+  )
+  default = []
+}
+
+##################################################################################
+
+
+##################################################################################
+# F5 Config Variables
+##################################################################################
+
+variable "default_route_interface" {
+  description = "The F5 BIG-IP interface name for the default route. Leave null to auto assign."
+  type        = string
+  default     = null
+}
+
+
+variable "hostname" {
+  description = "The F5 BIG-IP hostname"
+  type        = string
+  default     = "f5-ve-01"
+}
+
+variable "domain" {
+  description = "The F5 BIG-IP domain name"
+  type        = string
+  default     = "local"
+}
+
+variable "tmos_admin_password" {
+  description = "admin account password for the F5 BIG-IP instance"
+  type        = string
+  sensitive   = true
+}
+
+##################################################################################
+
+
+##################################################################################
+# A&O Declaration Sources
+##################################################################################
+
+variable "license_type" {
+  description = "License, may be 'none','byol','regkeypool','utilitypool'"
+  type        = string
+  default     = "none"
+
+  validation {
+    error_message = "License type may be one of 'none','byol','regkeypool','utilitypool'."
+    condition     = contains(["none", "byol", "regkeypool", "utilitypool"], var.license_type)
+  }
+}
+
+variable "byol_license_basekey" {
+  description = "Bring your own license registration key for the F5 BIG-IP instance"
+  type        = string
+}
+
+variable "license_host" {
+  description = "BIGIQ IP or hostname to use for pool based licensing of the F5 BIG-IP instance"
+  type        = string
+}
+
+variable "license_username" {
+  description = "BIGIQ username to use for the pool based licensing of the F5 BIG-IP instance"
+  type        = string
+}
+
+variable "license_password" {
```

Comment:
All passwords and tokens should be marked sensitive to help prevent accidental leaks